build.gradle.kts

```diff
@@ -49,81 +20,158 @@ allprojects {
 java {
-  sourceCompatibility = JavaVersion.toVersion(javaVersion)
-  targetCompatibility = JavaVersion.toVersion(javaVersion)
+  sourceCompatibility = javaCompatibilityVersion
+  targetCompatibility = javaCompatibilityVersion
 }
 
-intellij {
-  pluginName.set(name)
-  // This adds nullability assertions, but also compiles forms.
-  instrumentCode.set(true)
-  updateSinceUntilBuild.set(false)
-  version.set(ideVersion)
-  downloadSources.set(false)
-  val pluginList = mutableListOf(
-    project(":flutter-idea"), "java", "properties",
-    "junit", "Git4Idea", "Kotlin", "gradle",
-    "Groovy", "Dart:$dartVersion")
+dependencies {
+  intellijPlatform {
+    if (ideaProduct == "android-studio") {
+      create(IntelliJPlatformType.AndroidStudio, ideaVersion)
+    } else {//if (ide == "ideaIC") {
+      create(IntelliJPlatformType.IntellijIdeaCommunity, ideaVersion)
+    }
+    testFramework(TestFrameworkType.Platform)
+    val bundledPluginList = mutableListOf(
+      "com.intellij.java",
+      "com.intellij.properties",
+      "JUnit",
+      "Git4Idea",
+      "org.jetbrains.kotlin",
+      "org.jetbrains.plugins.gradle",
+      "org.intellij.intelliLang",
+    )
+    if (ideaProduct == "android-studio") {
+      bundledPluginList.add("org.jetbrains.android")
+      bundledPluginList.add("com.android.tools.idea.smali")
+    }
+    val pluginList = mutableListOf("Dart:$dartPluginVersion")
+    if (ideaProduct == "IC") {
+      pluginList.add("org.jetbrains.android:$androidPluginVersion")
+    }
 
-  // If 2023.3+ and IDEA (not AS), then "org.jetbrains.android:$androidVersion", otherwise "org.jetbrains.android",
-  // see https://github.com/flutter/flutter-intellij/issues/7145
-  if(ide == "android-studio") {
-    pluginList.add("org.jetbrains.android")
-  } else if (ide == "ideaIC") {
-    pluginList.add("org.jetbrains.android:$androidVersion")
-  }
+    // Finally, add the plugins into their respective lists:
+    // https://plugins.jetbrains.com/docs/intellij/plugin-dependencies.html#project-setup
+    bundledPlugins(bundledPluginList)
+    plugins(pluginList)
 
-  if (ide == "android-studio") {
-    pluginList.add(smaliPlugin)
-  }
-  pluginList.add(langPlugin)
-  if (ide == "android-studio") {
-    type.set("AI")
-    pluginList += listOf(project(":flutter-studio"))
+    // The warning that "instrumentationTools()" is deprecated might be valid, however, this error is produced by Gradle IJ plugin version
+    // 2.1.0 if this isn't included:
+    //  Caused by: org.gradle.api.GradleException: No Java Compiler dependency found.
+    //  Please ensure the `instrumentationTools()` entry is present in the project dependencies section along with the `intellijDependencies()` entry in the repositories section.
+    //  See: https://plugins.jetbrains.com/docs/intellij/tools-intellij-platform-gradle-plugin-dependencies-extension.html
+    instrumentationTools()
+//    pluginVerifier()
   }
-  plugins.set(pluginList)
 }
+//intellijPlatform {
```

Comment:
Do you want to commit this commented bit?

Reply:
100% we want to commit this, this is all related to getting the verifier working, see the TODO in this file and in `flutter-idea/build.gradle.kts`